src/systems/filecoin_blockchain/struct/chain/chain.go

```diff
@@ -18,8 +18,17 @@ func (chain *Chain_I) TipsetAtEpoch(epoch abi.ChainEpoch) Tipset {
 
 // Draws randomness from the tipset at or immediately prior to `epoch`.
 func (chain *Chain_I) RandomnessAtEpoch(epoch abi.ChainEpoch) abi.RandomnessSeed {
-	ts := chain.TipsetAtEpoch(epoch)
-	return ts.MinTicket().DrawRandomness(epoch)
+	if epoch < genesis {
+		genesisTS := chain.TipsetAtEpoch(genesis)
```

Comment:
When processing any messages in the genesis tipset (block) itself, there is no prior tipset to find here.

Reply:
By this I mean tipset made of the genesis block. Will comment accordingly